### PR TITLE
New post: Environment files and finding a bug in the ECS Agent

### PR DESCRIPTION
**Pull request by sapenroo**

### CODE REVIEW

**Comment by sapenroo** [comment]
Docs link if you want to review there:
https://docs.google.com/document/d/1AaIoymdV7xcoiF6FenFjGNDlFWM-ZnGkEtE96EDhxIs/edit?usp=sharing

File: _posts/2023-03-08-ecs-environment-files.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+layout: post
+title:  "Environment files and finding a bug in the ECS Agent"
+authors:
+  - "Simon Apen-Sadler"
+excerpt: >
+  How Deliveroo passes environment variables to containers in production
+
```
Comment (jamietanna) [inline _posts/2023-03-08-ecs-environment-files.md:8]: 🛑 we'll need to add a `date` and update the date of the post before we merge

**Review by martingolder — APPROVED** [review]
Approving the content does not contain anything I would deem a security risk

File: _posts/2023-03-08-ecs-environment-files.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+## Table of Contents
+{:.no_toc}
+
+1. Automatic Table of Contents Here
+{:toc}
+
+## Background
+
+We use the [Amazon Web Services][aws] (AWS) cloud provider to host our containers, primarily with the [Amazon Elastic Container Service][ecs] (ECS). As part of Deliveroo's [release process][app-dev], we use Hopper to deploy changes into production. We use environment variables to configure some of our applications. Hopper stores environment variables in its database (and links to [AWS Systems Manager Parameter Store][param-store] for sensitive values). During a deployment to ECS, Hopper generates a new [ECS Task definition][task-def] and registers it with ECS, inserting environment variables as necessary. Hopper then uses this new task definition to update the service in ECS.
```
Comment (Crunch09) [inline _posts/2023-03-08-ecs-environment-files.md:19]: I'm not sure this is explained somewhere else but don't think an external audience will have any idea what Hopper is? Maybe there should be a section to explain it before going into more detail here?

**Review by jamietanna — APPROVED** [review]

**Pull request merged**